File: 30-Days-of-Code/Day-6-Lets-Review/Elixir/solution.ex
defmodule Solution do
    def stringprint(x) do
        st = x |> String.split("", trim: true)
        firstHalf = st |> Enum.take_every(2)
        secondHalf = st |> Enum.drop(1) |> Enum.take_every(2)
        IO.puts "#{firstHalf} #{secondHalf}"
    end
end

n = IO.gets("") |> String.strip |> String.to_integer
m = for _ <- 1..n do
        IO.gets("") |> String.strip
end

#m = IO.stream(:stdio, :line) |> Enum.map(&String.strip/1)
Enum.each(m, &(Solution.stringprint(&1)))

File: 30-Days-of-Code/Day-6-Lets-Review/Ruby/solution.rb
n = gets.strip.to_i
words = Array.new(n)

(0..n-1).each do |i|
    words[i] = gets.strip
end

def process(word)
    letters = word.split("")
    firstHalf = []
    secondHalf = []
    letters.each_with_index do |l, i|
        i.even? ? firstHalf << l : secondHalf << l
    end
    
    "#{firstHalf.join} #{secondHalf.join}"
end

words.each do |word|
    puts process(word)
end
